Move protobuf settings for docs tests

javiermtorres opened the pull request.

javiermtorres commented:
@njbrake wdyt?

njbrake reviewed (changes requested):
Please put it in the docs tests, since this only applies for the docs. Also, before this is merged please run the "cookbook"' tests to ensure that the cookbook tests are passing, since I see a failed cookbook test run on main. Thanks!

javiermtorres commented:
> Please put it in the docs tests, since this only applies for the docs.

Unfortunately, this does not seem possible :-/ The issue lies in https://github.com/mozilla-ai/any-agent/blob/dbad5d5e65704f1c966009f2b429af4e5fb13f23/tests/conftest.py#L18
At that line, the otel sdk will be imported, so we cannot set the env var later than this line (which would be the case if we used a separate conftest). Do you think we can modify the import order for our conftest files?

njbrake commented:
@javiermtorres can you confirm that this change is required? I'm running the docs test without that env var set and everything is fine. 

<img width="823" alt="Screenshot 2025-06-30 at 5 56 43 AM" src="https://github.com/user-attachments/assets/79b27b2e-7f97-4238-afcf-3c33df16b959" />

javiermtorres commented:
> I am ok with it staying in the conftest but first please confirm that this env var is truly needed.

Let me take a closer look at this, yes. It could be that:

* The newest A2A SDK changed the protobuf version so it's not colliding with the version in otel
* We removed the otel examples in docs (we removed the cookbook, at least)

I'll keep you posted :)

njbrake reviewed (changes requested):
Please also fix the comment above the removal of the env var, and fix the docs at 
`docs/tracing.md` in the `## Adding an OpenTelemetry exporter` section

javiermtorres commented:
Re: the docs, I thought about them. I'd rather keep them as they are, since they state "it may be necessary", which is still true and will help the user in case of issues.
